### PR TITLE
[terra-dropdown-button] Add support for icons in split button

### PR DESCRIPTION
**Pull request by sycombs**
### Summary

This PR updates the `terra-dropdown-button` `SplitButton` to support icon-only, left-aligned icon, and right-aligned icon buttons in order to maintain passivity for the Fusion passthrough.

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

Screenshots:

image

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10337 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-dropdown-button/package.json [hunk 1]
```diff
@@ -57,5 +57,8 @@
     "LICENSE",
     "NOTICE",
     "README.md"
-  ]
+  ],
+  "devDependencies": {
+    "terra-icon": "^3.60.0"
```
Comment (sycombs) [inline packages/terra-dropdown-button/package.json:62]: Added this as a devDependency for use in Jest tests

File: packages/terra-dropdown-button/tests/jest/__snapshots__/SplitButton.test.jsx.snap [hunk 2]
```diff
@@ -105,7 +105,11 @@ exports[`Dropdown Button correctly applies the theme context className 1`] = `
             onKeyUp={[Function]}
             type="button"
           >
-            Primary Option
+            <span
+              className=""
```
Comment (sycombs) [inline packages/terra-dropdown-button/tests/jest/__snapshots__/SplitButton.test.jsx.snap:109]: Not sure if we want to leave the className attribute blank here or just remove it entirely.

**Comment by mjpalazzo** [comment]
@sycombs - The split button looks good and navigation is working. I observed on issue with JAWS where, when I Tab to the first button in the group, JAWS does not announce the name, role, value. 
For the rest of the components in the composite JAWS works as expected: JAWS announces the name role value and helper text of the menu button and the expanded menu.

File: packages/terra-dropdown-button/tests/wdio/__snapshots__/reference/orion-fusion-theme/en/chrome_medium/split-button-spec/text_first.png
[the remark this answers is not in the document]
Comment (sycombs) [reply packages/terra-dropdown-button/tests/wdio/__snapshots__/reference/orion-fusion-theme/en/chrome_medium/split-button-spec/text_first.png:1]: oops, good catch! fixed in [40003a0](https://github.com/cerner/terra-core/pull/4080/commits/40003a0e4980df3bbf9db1ba25586e47bb766c59)

**Comment by mjpalazzo** [comment]
> @sycombs - The split button looks good and navigation is working. I observed on issue with JAWS where, when I Tab to the first button in the group, JAWS does not announce the name, role, value. For the rest of the components in the composite JAWS works as expected: JAWS announces the name role value and helper text of the menu button and the expanded menu.

I rebooted by desktop and JAWS has returned to normal. The Split Button is working as required. Great work!

File: packages/terra-dropdown-button/tests/wdio/split-button-spec.js [hunk 3]
```diff
@@ -331,6 +331,26 @@ Terra.describeViewports('Split Button', ['medium'], () => {
     });
   });
 
+  describe('With icon', () => {
+    it('displays a button with icon first', () => {
+      browser.url('/raw/tests/cerner-terra-core-docs/dropdown-button/left-icon-split-button');
```
Comment (sdadn) [inline packages/terra-dropdown-button/tests/wdio/split-button-spec.js:336]: All these example seem pretty small, so I wonder if we can optimize these tests by having them all on the same page. Unless if it's not worth the effort.

Comment (sycombs) [reply packages/terra-dropdown-button/tests/wdio/split-button-spec.js]: sure, combined them into one test/snapshot here: [7396391](https://github.com/cerner/terra-core/pull/4080/commits/73963913e58237f8cedd81c14e5bef5df5e2b5e9)

Comment (sdadn) [reply packages/terra-dropdown-button/tests/wdio/split-button-spec.js]: That looks good! Can you also add text labels for easier identification? 

e.g.
```jsx
<h3>icon left</h3>
<iconLeftExample

<h3>icon right</h3>
<iconRightexample
```

Comment (sycombs) [reply packages/terra-dropdown-button/tests/wdio/split-button-spec.js]: Updated: [f66ce00](https://github.com/cerner/terra-core/pull/4080/commits/f66ce0032598bde290504304cf77b268ad95433d)

File: packages/terra-dropdown-button/tests/jest/__snapshots__/SplitButton.test.jsx.snap (hunk 2 above)
Comment (adoroshk) [reply packages/terra-dropdown-button/tests/jest/__snapshots__/SplitButton.test.jsx.snap:109]: NIT: I'd probably removed it for cleaner look, unless I am missing something. What is a benefit of keeping it?

**Review by adoroshk — APPROVED** [review]

File: packages/terra-dropdown-button/tests/jest/__snapshots__/SplitButton.test.jsx.snap (hunk 2 above)
Comment (sycombs) [reply packages/terra-dropdown-button/tests/jest/__snapshots__/SplitButton.test.jsx.snap:109]: No benefit, and I agree with you on the cleaner look. I'll remove it.

Comment (sycombs) [reply packages/terra-dropdown-button/tests/jest/__snapshots__/SplitButton.test.jsx.snap]: removed: [def249f](https://github.com/cerner/terra-core/pull/4080/commits/def249fad0186f93ce946799f1eff6bc01c74449)

**Review by sdadn — APPROVED** [review]

**Pull request merged**